src/java.desktop/windows/native/libawt/windows/awt_Window.cpp

```diff
@@ -1419,17 +1428,17 @@ BOOL AwtWindow::UpdateInsets(jobject insets)
             LONG style = GetStyle();
             if (style & WS_THICKFRAME) {
                 m_insets.left = m_insets.right =
-                    ::GetSystemMetrics(SM_CXSIZEFRAME);
+                    ::GetSystemMetrics(SM_CXSIZEFRAME) +
+                     extraPaddedBorderInsets;
                 m_insets.top = m_insets.bottom =
-                    ::GetSystemMetrics(SM_CYSIZEFRAME);
+                    ::GetSystemMetrics(SM_CYSIZEFRAME) +
+                     extraPaddedBorderInsets;
             } else {
                 m_insets.left = m_insets.right =
                     ::GetSystemMetrics(SM_CXDLGFRAME);
                 m_insets.top = m_insets.bottom =
                     ::GetSystemMetrics(SM_CYDLGFRAME);
```

Comment:
It's likely needed here as well.

Make the frame non-resizeable to get into this block.

Reply:
Added `SM_CXPADDEDBORDER` to Non-Resizable Frames. After the addition of padded border the second button is being shown similar to Resizable frames. But unlike resizable frames,  for the non-resizable frames - the actual frame size and the preferred size are different. Ideally the frame.getSize() should be equal to preferred size when frame.pack() is used. The cause of this issue seems to be different from the default insets value obtained from getSystemMetric().

Reply:
Side-by-Side comparison of Resizable and Non-Resizable frames  - Resizable frame slightly wider and taller compared to non-resizable. 
On UiScale =1,
Resizable Frame size= java.awt.Dimension[width=241,height=119]
Non-Resizable Frame size = java.awt.Dimension[width=237,height=115]

![image](https://user-images.githubusercontent.com/95945681/173165801-a387f853-bff0-4c16-ba83-056069137dac.png)

Reply:
Tested the` AwtWindow::UpdateInsets()` method for Resizable and Non-Resizable frame. 

-------------------------------------------------
**Non-Resizable UI Scale=1**

Outside Rect 0 56 0 198 
Inside Rect 0 0 0 176 
MenuBar Added
BEFORE Scale Down 73 9 9 9 
**AFTER Scale Down 73 9 9 9** 

Expected frame size: java.awt.Dimension[width=241,height=119]
Actual frame size: java.awt.Dimension[width=237,height=115]
Panel size: java.awt.Dimension[width=215,height=29]
**Frame Insets: java.awt.Insets[top=75,left=11,bottom=11,right=11]**

----------------------------------------------
**Non-Resizable** 

Outside Rect 0 56 0 198 
Inside Rect 0 0 0 176 
MenuBar Added
BEFORE Scale Down 73 9 9 9 
**AFTER Scale Down 49 6 6 6** 

Expected frame size: java.awt.Dimension[width=231,height=90]
Actual frame size: java.awt.Dimension[width=229,height=88]
Panel size: java.awt.Dimension[width=215,height=31]
**Frame Insets: java.awt.Insets[top=50,left=7,bottom=7,right=7]**

----------------------------------------------
**Resizable UI Scale=1**

Outside Rect 0 56 0 198 
Inside Rect 0 0 0 176 
MenuBar Added 
BEFORE Scale Down 75 11 11 11 
**AFTER Scale Down 75 11 11 11** 

Expected frame size: java.awt.Dimension[width=241,height=119]
Actual frame size: java.awt.Dimension[width=241,height=119]
Panel size: java.awt.Dimension[width=219,height=33]
**Frame Insets: java.awt.Insets[top=75,left=11,bottom=11,right=11]**

----------------------------------------------
**Resizable** 

Outside Rect 0 56 0 198 
Inside Rect 0 0 0 176 
MenuBar Added
BEFORE Scale Down 75 11 11 11 
**AFTER Scale Down 50 7 7 7** 

Expected frame size: java.awt.Dimension[width=231,height=90]
Actual frame size: java.awt.Dimension[width=231,height=90]
Panel size: java.awt.Dimension[width=217,height=33]
**Frame Insets: java.awt.Insets[top=50,left=7,bottom=7,right=7]**

-------------------------------------------------------------

Based on these logs, I observed that the final frame insets for non-resizable frame obtained by frame.getInsets() does not match with the scaled down insets where as for resizable frame it does. 

**Clarification Point** - should the preferred size of a resizable and non-resizable frame be the same when frame.pack() ? The default insets obtained for both are from different system metrics  - **::GetSystemMetrics(SM_CYSIZEFRAME) &  ::GetSystemMetrics(SM_CYDLGFRAME)** respectively.

Reply:
> **Clarification Point** - should the preferred size of a resizable and non-resizable frame be the same when frame.pack() is used? The default insets obtained for both are from different system metrics - **::GetSystemMetrics(SM_CYSIZEFRAME) & ::GetSystemMetrics(SM_CYDLGFRAME)** respectively.

Do `::GetSystemMetrics(SM_CYSIZEFRAME)` and `::GetSystemMetrics(SM_CYDLGFRAME)` return the same value? If yes, then I expect the size of the frame should be the same for resizable and non-resizable case.

Let's put it another way, I expect the `pack()` makes the panel size to its preferred size.

In either case, the `frame.getPreferredSize()` returns the same result but the actual size differs.

I don't consider it a blocker for this integration: the resizable case is handled perfectly now; the non-resizable case also ensures the second button is visible now.

However, I think we should find out what contributes to the difference in the actual size under a separate bugid.

Reply:
@aivanov-jdk  Thank you for clarifying.  Yes ::GetSystemMetrics(SM_CYSIZEFRAME) and ::GetSystemMetrics(SM_CYDLGFRAME) return different values, thus we get different actual frame sizes for resizable and non-resizable frames.

As you rightly pointed out **_: the frame.getPreferredSize() returns the same result but the actual size differs for Non-Resizable frame._**

As suggested I will create a new bug to track actual and preferred size difference for non-resizable frame.

Reply:
> @aivanov-jdk Thank you for clarifying. Yes ::GetSystemMetrics(SM_CYSIZEFRAME) and ::GetSystemMetrics(SM_CYDLGFRAME) return different values, thus we get different actual frame sizes for resizable and non-resizable frames.

If these are different, why does the preferred size remain the same?

If these parameters return different values, the preferred size should be different.

> As you rightly pointed out **_: the frame.getPreferredSize() returns the same result but the actual size differs for Non-Resizable frame._**
> 
> As suggested I will create a new bug to track actual and preferred size difference for non-resizable frame.

Thank you!

Reply:
> However, I think we should find out what contributes to the difference in the actual size under a separate bugid.

Tracked by [JDK-8288325](https://bugs.openjdk.org/browse/JDK-8288325)